mobile/src/main/java/rs/readahead/washington/mobile/views/fragment/MicFragment.kt

```diff
@@ -41,11 +158,308 @@ class MicFragment : Fragment() {
         // TODO: Rename and change types and number of parameters
         @JvmStatic
         fun newInstance(param1: String, param2: String) =
-                MicFragment().apply {
-                    arguments = Bundle().apply {
-                        putString(ARG_PARAM1, param1)
-                        putString(ARG_PARAM2, param2)
-                    }
+            MicFragment().apply {
+                arguments = Bundle().apply {
+                    putString(ARG_PARAM1, param1)
+                    putString(ARG_PARAM2, param2)
                 }
+            }
+    }
+
+    override fun onStart() {
+        super.onStart()
+        activity.startLocationMetadataListening()
+    }
+
+    override fun onStop() {
+        activity.stopLocationMetadataListening()
+
+        super.onStop()
+    }
+
+    override fun onDestroy() {
+        animator!!.end()
```

Comment:
app can crash here if animator is null, maybe use null safe check `animator?.end()`